src/lib/docker-containers.mjs

```diff
@@ -0,0 +1,324 @@
+import { container } from "../cli.mjs";
+import { CommandError } from "./errors.mjs";
+
+/**
+ * Ensures the container is running
+ * @param {string} imageName The name of the image to create the container from
+ * @param {string} containerName The name of the container to start
+ * @param {number} hostPort The port on the host machine mapped to the container's port
+ * @param {number} containerPort The port inside the container Fauna listens on
+ * @param {boolean} pull Whether to pull the latest image
+ * @returns {Promise<void>}
+ */
+export async function ensureContainerRunning({
+  imageName,
+  containerName,
+  hostPort,
+  containerPort,
+  pull,
+}) {
+  const logger = container.resolve("logger");
+  if (pull) {
+    await pullImage(imageName);
+  }
+  const logStream = await startContainer({
+    imageName,
+    containerName,
+    hostPort,
+    containerPort,
+  });
+  logger.stderr(
+    `[StartContainer] Container '${containerName}' started. Monitoring HealthCheck for readiness.`,
```

Comment:
FWIW, we could update `colorize` (and `codeToAnsi`) to support colorizing logs in a couple lines:

https://textmate-grammars-themes.netlify.app/?theme=github-dark-high-contrast&grammar=log

Reply:
That's a good idea.